### PR TITLE
Tasks/setup json schema

### PR DESCRIPTION
**Pull request by npauzenga**
Hey @Enriikke mind checking out an initial pass of the `User` schema? It includes some pre-emptive switching to a new profiles controller on `#show` and `#index` which is not currently supported. 

How do you like the way my JSON objects are formatted? A user in my app looks something like this --

``` json
{
  "first_name": "Nate",
  "last_name": "Pauzenga",
  "email": "test@admin.com",
  "city": "Annapolis",
  "state": "Maryland",
  "country": "USA",
  "password": "helloworld"
}
```

### CODE REVIEW

**Comment by npauzenga** [comment]
Hey @Enriikke you'd mentioned on our call tonight that we're deviating a little from the JSON API spec. I know that's totally fine and I don't want to be precious about it but would you mind elaborating? The schema we're using is derived from the JSON API specification (http://jsonapi.org/faq/#is-there-a-json-schema-describing-json-api) but it looks like the structure is a little different than the spec?

File: config/schema/api.json [hunk 1]
```diff
@@ -0,0 +1,1012 @@
...
+        {
+          "description": "Delete an existing game_night.",
+          "href": "/game_nights/{(%23%2Fdefinitions%2Fgame_night%2Fdefinitions%2Fidentity)}",
+          "method": "DELETE",
+          "rel": "destroy",
+          "http_header": {
+            "Authorization": "Bearer: iOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE0NTM5MzkwMjYsImF1ZCI6bnVsbCwic3ViIjoxfQ.TSgJie1vAWxR_ym_uazjKs43rpQ50OKmb2KZ2N6ALs4"
+          },
+          "title": "Delete"
+        },
```
Comment (npauzenga) [inline config/schema/api.json:135]: @Enriikke so for deleting records, the response we're going for is 204/no-content. I'm not entirely sure how to represent this though. Currently it just looks for all of the properties on the appropriate model --

``` json
{
  "created_at": "2015-01-01T12:00:00Z",
  "id": 1294,
  "location_name": "Silver Spring Library",
  "location_address": "8401 Fenton st. Silver Spring, MD 20910",
  "time": "2013-09-12 22:49:27 +0530",
  "updated_at": "2015-01-01T12:00:00Z"
}
```

**Comment by npauzenga** [comment]
@Enriikke so we've got schema for all of our AR models now. I'm curious - would you produce a separate schema for things like `password resets`? They're sending JSON messages back to the client (though maybe they shouldn't be?) so I'd think we would want to validate their format...

Also, I should do schema for the `else` branches on the controllers. I'm rendering errors and I'd like them to conform. I'm looking around for ways to represent this so perhaps I'll beat you to it but do you have suggestions?

File: config/schema/api.json [hunk 2]
```diff
@@ -0,0 +1,1012 @@
...
+        {
+          "description": "Info for existing game_night.",
```
Comment (npauzenga) [inline config/schema/api.json:137]: @Enriikke I believe the `id` param is implied but should I specify it as a parameter for the request?

Comment (enriikke) [inline config/schema/api.json:137]: You only need to specify it as a portion of the request url. If you notice the line below you are already doing that with this beautifully legible string `{(%23%2Fdefinitions%2Fgame_night%2Fdefinitions%2Fidentity)}` :smile:

File: config/schema/api.json (hunk 1 above)
Comment (enriikke) [inline config/schema/api.json:135]: What I've been doing is defining the link without a target schema (which is what defines the response). The way you have it right now should work, are you getting any errors?

**Comment by enriikke** [comment]
Awesome work @npauzenga!! :shipit: :rocket: :baby_chick: 

Your resource schemas look good. I think I would also implement a schema for passwords since we are treating them as a resource during the reset flow. I haven't been validating my error responses because they have been evolving a bit but is probably something I should :sweat_smile:. You can use `oneOf` in the schema to define both success and error responses and then the validator can check that is either one of those and nothing more.

As for jsonapi, we are using flat json when we render our models. That just means that they are serialized with all their attributes at the root and nothing fancy is going on. jsonapi defines a different way to serialize things for example the a user would look like:

``` js
{
  "data": {
    "id": 1,
    "type": "users",
    "attributes": {
      "first_name": "Nate",
      "last_name": "Pauzenga",
      "email": "test@admin.com",
      "city": "Annapolis",
      "state": "Maryland",
      "country": "USA",
      "password": "helloworld" // probably shouldn't even be here (no plain-text) :-)
    }
  }
}
```

jsonapi also defines a way to serialize an object's relationships, where in flatjson we would just add them with the rest of the properties.

P.S.: The jsonapi schema was a great find!! I wasn't using it (didn't even know it existed) so I was manually defining it myself. Now I know what I will be working on next week :smiley:

**Pull request merged**